paparazzi-gradle-plugin/src/main/java/app/cash/paparazzi/gradle/PrepareResourcesTask.kt

```diff
@@ -51,18 +54,50 @@ open class PrepareResourcesTask : DefaultTask() {
   @get:PathSensitive(PathSensitivity.RELATIVE)
   internal val platformDataRoot: DirectoryProperty = project.objects.directoryProperty()
 
+  @get:Input
+  internal val nonTransitiveRClassEnabled: Property<Boolean> =
+    project.objects.property(Boolean::class.java)
+
+  /**
+   * This is the "official" input for wiring task dependencies correctly, but is otherwise
+   * unused.
+   */
+  @PathSensitive(PathSensitivity.NAME_ONLY)
+  @InputFiles
+  internal fun getPackageAwareRArtifactFiles(): FileCollection =
+    resolvedArtifactResults.artifactFiles
+
   @get:OutputFile
   internal val paparazziResources: RegularFileProperty = project.objects.fileProperty()
 
+  internal fun setPackageAwareRArtifacts(resources: ArtifactCollection) {
+    this.resolvedArtifactResults = resources
+  }
+
+  private lateinit var resolvedArtifactResults: ArtifactCollection
+
   private val projectDirectory = project.layout.projectDirectory
 
   @TaskAction
   fun writeResourcesFile() {
     val out = paparazziResources.get().asFile
     out.delete()
+
+    val mainPackage = packageName.get()
+    val resourcePackageNames = if (nonTransitiveRClassEnabled.get()) {
+      resolvedArtifactResults.mapNotNull { resolvedArtifactResult ->
+        resolvedArtifactResult.file.useLines { lines -> lines.first() }
+      }
+        .toMutableList()
+        .apply { add(0, mainPackage) }
+        .joinToString(",")
+    } else {
+      mainPackage
+    }
```

Comment:
maybe we can clean this up like so: https://github.com/cashapp/paparazzi/pull/275/files#diff-86798d2cbcaa4785e1f3873d6246b6acd557a1bae3d0cae7996d3fe7242066d3

Reply:
The new approach does not supply the main package name, so it will have to be just a little bit different.

How about this? 
```
val mainPackage = packageName.get()
var resourcePackageNames = mainPackage
if (nonTransitiveRClassEnabled.get()) {
  resourcePackageNames += artifactFiles.filterNotNull()
    .takeIf { it.isNotEmpty() }
    ?.joinToString(",", ",") { file ->
      file.useLines { it.first() }
    } ?: ""
}
```

Reply:
Or this? This one is my preference.
```
val mainPackage = packageName.get()
var resourcePackageNames = mainPackage
if (nonTransitiveRClassEnabled.get()) {
  resourcePackageNames += artifactFiles
    .mapNotNull { file -> file.useLines { it.first() } }
    .takeIf { it.isNotEmpty() }
    ?.joinToString(",", ",")
    ?: ""
}
```

Reply:
For completeness, here's the cleaned up version of the current way:
```
val mainPackage = packageName.get()
val resourcePackageNames = if (nonTransitiveRClassEnabled.get()) {
  artifactFiles.mapNotNull { it.useLines { lines -> lines.first() } }
    .toMutableList()
    .apply { add(0, mainPackage) }
    .joinToString(",")
} else {
  mainPackage
}
```